xbmc/Application.cpp

```diff
@@ -2296,11 +2296,18 @@ bool CApplication::OnAction(const CAction &action)
       if (m_muted)
         UnMute();
       float volume = m_volumeLevel;
+      int volumesteps = CSettings::GetInstance().GetInt(CSettings::SETTING_AUDIOOUTPUT_VOLUMESTEPS);
+      // appliance.xml may contain invalid values. sanity check
```

Comment:
Do we really need to guard against that? An appliance doing something like this basically criples its own system which is certainly not in their interest. The only check really necessary would be for `volumesteps` to not be `0` to avoid a division by zero problem.